docs/TerraToolkitUpgradeGuide-v3.0.0.md

```diff
@@ -0,0 +1,121 @@
+# Terra Toolkit Upgrade Guide v3.0.0
+This document will provide information on upgrading from Terra Toolkit 2.x to 3.0.0.
+
+If you use terra-dev-site please also refer to it's upgrade guide as it abstracts away from additional config that is required here.
+
+## Webpack 4
+Terra toolkit and it's dependencies have been upgraded to consume webpack 4. Terra toolkit is now intended to provide Webpack to the libraries that consume it.
+Users of Terra toolkit should not add a direct dependency to webpack in their package JSON after updating. We have made this change to provide a single point of upgrading webpack.
+
+Terra toolkit now provides a default webpack config. This config will need to be referenced by the app level webpack config to add in an entry (unless consuming terra-dev-site). We recommend using [webpack-merge](https://github.com/survivejs/webpack-merge) to combine configs.
+The default webpack config is now a function that will flex between production and development modes by passing in the -p flag when compiling with webpack. For more about webpack config as a function go [here](https://webpack.js.org/configuration/configuration-types/).
```

Comment:
See webpack's documentation on [configuration types](https://webpack.js.org/configuration/configuration-types/) for more information.